Add MySQL/MariaDB as a ROOT system dep on CC7

dberzano opened the pull request:
Enables building of MySQL ROOT classes.

dberzano commented:
@Barthelemy this is all we need, along with alisw/docks#155.

Note that this PR is to master as adding system dependencies does not trigger rebuilds. Current production ROOT tag will not be rebuilt by merging this.

Barthelemy commented:
Could you remind me what is the policy for the branches in alidist ? master ? dev ? prod/IB/... ? 
Thank you

dberzano commented:
Hi, we have changed it since we've moved AliRoot to GitHub.

* Main branch is `master`
* Development branch is `dev` (open your pull requests to `dev`, not to `master`, apart from exceptions, see below)

Changes in recipes that affect production AliRoot/AliPhysics (_i.e._ trigger a rebuild) should never go to `master` but to `dev`. In this case, as said, this change does not trigger a rebuild, therefore it was safe to merge it to `master`.

When in doubt, open a PR to `dev`.
